packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/KeyValuesTable.test.js

```diff
@@ -68,96 +90,122 @@ describe('<KeyValuesTable>', () => {
       value: JSON.stringify(['name=mos-def', 'code=12345']),
       expected: 'name=mos-def, code=12345',
     },
-    // render().text() does not preserve full escaping of rendered JSON,
-    // so instead rely on jest snapshot comparison.
-    // Key observations:
-    // - "<xss>" key is encoded as &lt;xss&gt;
-    // - link value is rendered as <a>
-    // - xss_link value is rendered as plain string
     { key: 'jsonkey', value: JSON.stringify(jsonValue), snapshot: true },
   ];
 
-  beforeEach(() => {
-    wrapper = shallow(<KeyValuesTable data={data} />);
-  });
+  it('renders KeyValueTable container with correct structure', () => {
+    render(<KeyValuesTable data={data} />);
+
+    const container = document.querySelector('.KeyValueTable.u-simple-scrollbars');
+    expect(container).toBeInTheDocument();
 
-  it('renders without exploding', () => {
-    expect(wrapper).toBeDefined();
-    expect(wrapper.find('.KeyValueTable').length).toBe(1);
+    const table = screen.getByRole('table');
+    expect(table).toHaveClass('u-width-100');
   });
 
-  it('renders a table row for each data element', () => {
-    const trs = wrapper.find('tr');
-    expect(trs.length).toBe(data.length);
-    trs.forEach((tr, i) => {
-      expect(tr.find('.KeyValueTable--keyColumn').text()).toMatch(data[i].key);
+  it('renders table row for each data element with correct key columns', () => {
+    render(<KeyValuesTable data={data} />);
+
+    const rows = screen.getAllByRole('row');
+    expect(rows).toHaveLength(data.length);
+
+    data.forEach(item => {
+      const keyCell = screen.getByText(item.key);
+      expect(keyCell).toHaveClass('KeyValueTable--keyColumn');
     });
   });
 
-  it('renders the expected text for each span value', () => {
-    const el = wrapper.find('.ub-inline-block');
-    expect(el.length).toBe(data.length);
-    el.forEach((valueDiv, i) => {
+  it('renders expected text content for each span value', () => {
+    render(<KeyValuesTable data={data} />);
+
+    const valueElements = document.querySelectorAll('.ub-inline-block');
+    expect(valueElements).toHaveLength(data.length);
+
+    valueElements.forEach((valueDiv, i) => {
       if (data[i].expected) {
-        expect(valueDiv.render().text()).toBe(data[i].expected);
+        expect(valueDiv).toHaveTextContent(data[i].expected);
       } else if (data[i].snapshot) {
-        expect(valueDiv).toMatchSnapshot();
```

Comment:
I don't think this is equivalent. If we're not using snapshot then the `expected` parameter of the test should contain the desired rendered value.